### PR TITLE
Add check.local_ip config option

### PR DESCRIPTION
**Pull request by analogic**
Changes proposed in this pull request:
- add check.local_ip flag to allow skipping scaning localhost emails

(depends on https://github.com/haraka/Haraka/pull/2532)

Checklist:
- [x] docs updated
- [x] tests updated
- [x] Changes.md updated
- [x] package.json.version bumped
- [ ] published to NPM (will be done by @core)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -260,6 +261,8 @@ exports.wants_skip = function (connection) {
 
     if (!plugin.cfg.check.authenticated && connection.notes.auth_user) return true;
 
+    if (connection.remote.is_local) return !plugin.cfg.check.local_ip;
```
Comment (msimerson) [inline index.js]: I'd prefer that this line of code not affect the flow except when its a positive hit. I find it much easier to reason about the flow this way:

```suggestion
    if (plugin.cfg.check.local_ip && connection.remote.is_local) return true;
```

Comment (msimerson) [reply index.js]: > this logic is hard to think through

Agreed! Does [this PR](https://github.com/analogic/haraka-plugin-rspamd/pull/1) help?  If not, feel free to take some liberties as I demonstrated in the PR. Perhaps a better name for the function. Or another type of flow control. Making the logic easier to reason about is welcome.

**Comment by analogic** [comment]
@msimerson awesome! LGTM now

**Comment by msimerson** [comment]
````
~/git/haraka-plugin-rspamd $ npm publish
npm notice 
npm notice 📦  haraka-plugin-rspamd@1.1.2
npm notice === Tarball Contents === 
npm notice 978B   package.json                    
npm notice 138B   .codeclimate.yml                
npm notice 439B   .eslintrc.json                  
npm notice 458B   .travis.yml                     
npm notice 241B   appveyor.yml                    
npm notice 467B   Changes.md                      
npm notice 12.7kB index.js                        
npm notice 1.1kB  LICENSE                         
npm notice 4.1kB  README.md                       
npm notice 337B   run_tests                       
npm notice 377B   .github/ISSUE_TEMPLATE.md       
npm notice 211B   .github/PULL_REQUEST_TEMPLATE.md
npm notice 381B   config/rspamd.ini               
npm notice 225B   test/.eslintrc.json             
npm notice 7.6kB  test/index.js                   
npm notice === Tarball Details === 
npm notice name:          haraka-plugin-rspamd                    
npm notice version:       1.1.2                                   
npm notice package size:  8.2 kB                                  
npm notice unpacked size: 29.7 kB                                 
npm notice shasum:        5af350381d493d52a2d7feed16d7f37ea2551d15
npm notice integrity:     sha512-uXM1ElQ6ixN3L[...]yxlFY+eK51RMw==
npm notice total files:   15                                      
npm notice 
+ haraka-plugin-rspamd@1.1.2
````

**Pull request merged**